modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments-widgets/components/TabsPanel.js

```diff
@@ -49,6 +63,16 @@ export default function TabsPanel({
 	const scrollPositionRef = useRef(scrollPosition);
 	scrollPositionRef.current = scrollPosition;
 
+	useEffect(() => {
+		if (activeTabId === COLLECTION_IDS.widgets && !widgets.length) {
+			dispatch(
+				loadWidgets({
+					fragmentEntryLinks: widgetFragmentEntryLinksRef.current,
+				})
+			);
+		}
+	}, [activeTabId, dispatch, widgetFragmentEntryLinksRef, widgets]);
```

Comment:
If the user doesn't have any portlet this will keep running over and over again isn't it?

Reply:
Um, I didn't think about that option 🤔